### ISSUE TITLE
Open shadowing warning mistriggers

### ISSUE THREAD

**Issue by rossberg** [ocaml/ocaml#12494]
Warning 44 (open shadowing) triggers in a case where no shadowing actually occurs. Minimal example:
```
$ ocaml -w +44
OCaml version 4.14.0
Enter #help;; for help.
# module M = struct let f () = 1 end;;
module M : sig val f : unit -> int end
# let f () = M.(f ());;
Warning 44 [open-shadow-identifier]: this open statement shadows the value identifier f (which is later used)
val f : unit -> int = <fun>
```
Note that the `let` is not recursive, so `f` should not be in scope on its r.h.s..

**Comment by gasche** [comment]
Good catch! I propose a fix in #12498.

**Comment by gasche** [comment]
This is now fixed in trunk, to be released in 5.2 (not the upcoming 5.1 release).
